src/clients/c++/experimental_api_v2/perf_client/inference_profiler.h

```diff
@@ -0,0 +1,403 @@
+  /// request rate nor concurrency. Requires load manager to be loaded with
+  /// a file specifying the time intervals.
+  /// \param summary Appends the measurements summary at the end of this list.
+  /// \param meets_threshold Returns whether the measurement met the threshold.
+  /// \return Error object indicating success
+  /// or failure.
+  nic::Error Profile(std::vector<PerfStatus>& summary, bool* meets_threshold);
+
+  /// A helper function for profiling functions.
+  /// \param clean_starts Whether or not to reset load cycle with every
+  /// measurement trials.
+  /// \param status_summary Returns the summary of the measurement.
+  /// \param is_stable Returns whether the measurement stabilized or not.
+  /// \return Error object indicating success or failure.
+  nic::Error ProfileHelper(
+      const bool clean_starts, PerfStatus& status_summary, bool* is_stable);
+
+  /// Helper function to perform measurement.
+  /// \param status_summary The summary of this measurement.
+  /// \return Error object indicating success or failure.
+  nic::Error Measure(PerfStatus& status_summary);
+
+  /// Gets the server side statistics
+  /// \param model_status Returns the status of the models provided by
+  /// the server. If the model being profiled is non-ensemble model,
+  /// only its status will be returned. Otherwise, the status of the composing
+  /// models will also be returned.
+  /// \return Error object indicating success or failure.
+  nic::Error GetServerSideStatus(
+      std::map<ModelIdentifier, ModelStatistics>* model_status);
+
+  // A helper fuction for obtaining the status of the models provided by the
+  // server.
+  nic::Error GetServerSideStatus(
+      ModelStatistics& server_status, const ModelIdentifier model_identifier,
+      std::map<ModelIdentifier, ModelStatistics>* model_status);
+
+  /// Sumarize the measurement with the provided statistics.
+  /// \param timestamps The timestamps of the requests completed during the
+  /// measurement.
+  /// \param start_status The model status at the start of the measurement.
+  /// \param end_status The model status at the end of the measurement.
+  /// \param start_stat The accumulated context status at the start.
+  /// \param end_stat The accumulated context status at the end.
+  /// \param summary Returns the summary of the measurement.
+  /// \return Error object indicating success or failure.
+  nic::Error Summarize(
+      const TimestampVector& timestamps,
+      const std::map<ModelIdentifier, ModelStatistics>& start_status,
+      const std::map<ModelIdentifier, ModelStatistics>& end_status,
+      const nic::InferStat& start_stat, const nic::InferStat& end_stat,
+      PerfStatus& summary);
+
+  /// A helper function to get the start and end of a measurement window.
+  /// \param timestamps The timestamps collected for the measurement.
+  /// \return the start and end timestamp of the measurement window.
+  std::pair<uint64_t, uint64_t> MeasurementTimestamp(
```

Comment:
const& or pass in a std::pair* to intialize, or if not performance critical say something in the comment ("returning by value because not performance critical")